### ISSUE TITLE
Error when activating charge

### ISSUE THREAD

**Issue by DavidVII** [nsweeting/shopify#23]
Hello, I am trying to activate a `RecurringApplicationCharge`. I am getting the following case clause error:

```
Shopify.RecurringApplicationCharge.activate(session, charge.id)
** (CaseClauseError) no case clause matching: nil
    (hackney) deps/hackney/src/hackney_request.erl:312: :hackney_request.handle_body/4
    (hackney) deps/hackney/src/hackney_request.erl:81: :hackney_request.perform/2
    (hackney) deps/hackney/src/hackney.erl:358: :hackney.send_request/2
    (httpoison) lib/httpoison/base.ex:439: HTTPoison.Base.request/9
    (shopify) lib/shopify/adapters/http.ex:13: Shopify.Adapters.HTTP.post/1
```

My session  works fine with other requests:

```elixir
%Shopify.Session{
  access_token: "mytoken",
  api_key: nil,
  client_id: nil,
  client_secret: nil,
  password: nil,
  shop_name: "myshop",
  type: :oauth
}
```

I can use the session and charge.id to call `Shopify.RecurringApplicationCharge.find/2` which returns the correct response, but when I try and call `activate` it errors out.

Has anyone else run into this issue as well?

**Comment by Ninigi** [comment]
I've come across this error before, iirc this is because hackney can not handle a `nil` value for a post/put body. I fixed that in recent commits, however I haven't been able to do a comprehensive test for those cases.

Can you test your usecase against the master branch? Setting up an application charge just for test cases is a PITA, so if you have everything set up already that would be hugely appreciated 🙏

**Comment by DavidVII** [comment]
Thanks, @Ninigi. I am in fact using the latest from the master branch. Is there any particular area of the codebase that comes to mind? I don't mind looking into this as well since I do have an app setup ATM. Let me know.

**Comment by Ninigi** [comment]
@DavidVII I just pushed the post body fix (30 mins ago?), so maybe you could update and try again? 😆

**Comment by DavidVII** [comment]
@Ninigi haha my bad. Ya, I'll try again. Thanks!

**Comment by Ninigi** [comment]
@DavidVII [this is the culprit](https://github.com/nsweeting/shopify/blob/master/lib/shopify/adapters/http.ex) (at least that's my guess for now)

**Comment by DavidVII** [comment]
@Ninigi yep looks like that's it. I was able to activate a charge. Interestingly enough, I got an error when deleting a charge. It did, in fact, cancel the charge, but it returned this:

```
 no case clause matching: {:error, :invalid, 0}
    (shopify) lib/shopify/response.ex:24: Shopify.Response.parse_json/2
    (shopify) lib/shopify/response.ex:12: Shopify.Response.new/3
```

That probably warrants a separate issue. I'll test it out and see if it's valid.

Thanks for your help, @Ninigi! I'll close this one out for now since it appears that 77fd3aaec782c3350b5c001f89fb8bf12137c919 addresses this issue.
